test/runtests.jl

```diff
@@ -40,6 +40,43 @@ deep_equal(a::T, b::S) where {T,S} = false
 const DATADIR = joinpath(@__DIR__, "data")
 
 @testset "Just Do It" begin
+
+    @testset "sprintf functionality" begin
+        # Moderate sized numbers
+        @test sprintf_G_under_8(0.123) == "0.123"
+        @test sprintf_F_under_8(0.123) == "0.123"
+
+        @test sprintf_G_under_8(1.123) == "1.123"
+        @test sprintf_F_under_8(1.123) == "1.123"
+
+        @test sprintf_G_under_8(10.123) == "10.123"
+        @test sprintf_F_under_8(10.123) == "10.123"
+
+        # Moderate numbers, many digits
+        @test sprintf_G_under_8(0.8945620050698592) == "0.894562"
+        @test sprintf_F_under_8(0.8945620050698592) == "0.894562"
+
+        # Large numbers / few digits
+        @test sprintf_G_under_8(0.123e10) == "1.23E+9"
+        # decimal version cannot handle it:
+        err = ErrorException("failed to fit number into EDF's 8 ASCII character limit: 1.23e9")
+        @test_throws err sprintf_F_under_8(0.123e10)
+
+        # Large numbers / many digits
+        @test sprintf_G_under_8(0.8945620050698592e10) == "8.946E+9"
+        err = ErrorException("failed to fit number into EDF's 8 ASCII character limit: 8.945620050698591e9")
+        @test_throws err sprintf_F_under_8(0.8945620050698592e10)
+
+        # Small numbers / few digits
+        @test sprintf_G_under_8(0.123e-10) == "1.23E-11"
+        # decimal version underflows:
+        @test sprintf_F_under_8(0.123e-10) == "0"
```

Comment:
I really feel like this is dangerous to do silently...zero is weird for all kinds of reasons.  can we `@warn` at least if we're not going to throw?

Reply:
Sure